airbyte-integrations/connectors/source-klaviyo/source_klaviyo/manifest.yaml

```diff
@@ -212,6 +235,115 @@ definitions:
       path: "templates"
       cursor_field: "updated"
 
+  campaigns_stream:
+    # Docs: https://developers.klaviyo.com/en/v2024-10-15/reference/get_campaigns
+    name: "campaigns"
+    $ref: "#/definitions/base_stream"
+    incremental_sync:
+      type: DatetimeBasedCursor
+      cursor_field: "updated_at"
+      start_datetime: "{{ config.get('start_date', '2012-01-01T00:00:00Z') }}"
+      datetime_format: "%Y-%m-%dT%H:%M:%S%z"
+      cursor_datetime_formats:
+        - "%Y-%m-%dT%H:%M:%S.%f%z"
+        - "%Y-%m-%dT%H:%M:%S%z"
+        - "%Y-%m-%d %H:%M:%S%z"
+      step: P60D
+      cursor_granularity: PT1S
+    retriever:
+      $ref: "#/definitions/base_retriever"
+      requester:
+        $ref: "#/definitions/requester"
+        request_parameters:
+          "filter": "and(greater-or-equal({{ parameters.cursor_field }},{{ stream_interval.start_time }}),less-or-equal({{ parameters.cursor_field }},{{ stream_interval.end_time }}),equals(messages.channel,'{{ stream_partition['campaign_type'] }}'),equals(archived,{{ stream_partition['archived'] }}))"
+          "sort": "{{ parameters.cursor_field }}"
+      partition_router:
+        - type: ListPartitionRouter
+          values:
```

Comment:
There is also `mobile_push` based on [this doc](https://developers.klaviyo.com/en/reference/get_campaigns). Should we add this here? I'm fine for this to be out-of-scope for now if we don't have indications that it is needed as this would trigger a full refresh of `mobile_push`+ `archived` and `mobile_push` + `not archived` and we would need to figure this out if we don't want this to happen (although maybe we want this to happen)

Reply:
you are referring to docs for api version 2025-01-15, but pr updates api to version 2024-10-15, for this version ([docs](https://developers.klaviyo.com/en/v2024-10-15/reference/get_campaigns)) we only have `email` and `sms` filters.

Reply:
Thanks for the precision!